### PR TITLE
Install latest Zsh+Zim

### LINKED ISSUE #59: Zim repo doesnt have Templates folder
In 1-basicsetup.sh line 73  we copy the .zshrc .zlogin and .zimrc from the the templates folder in the repo https://github.com/zimfw/zimfw. However i dont find any such folder in the repo's master branch and is present in another branch called **zsh-5.0** and **zsh-5.2** and hence while running the script it errors out saying that ,zshrc does not exist when we echo some lines on to that file.

### PR DESCRIPTION
**Pull request by rajat2004**
Closes #59 
https://github.com/zimfw/zimfw

TODO:
- [x] Actually test if this is working :sweat_smile: (Probably in VirtualBox)
- [x] Install latest `zsh` for Ubuntu 16.04 (Newer Zim requires Zsh >= 5.2)
- [x] Update instructions for multiple users

### CODE REVIEW

File: 1-BasicSetUp.sh [hunk 1]
```diff
@@ -67,15 +67,16 @@ if [[ -d $zsh_folder ]];then
 fi
 
 spatialPrint "Setting up Zsh + Zim now"
-execute sudo apt-get install zsh -y
+# Ubuntu 16.04 has older zsh in repository
+if [[ $(cat /etc/os-release | grep "VERSION_ID" | grep -o -E '[0-9][0-9]' | head -n 1) -le 16 ]]; then
+    sh -c "$(wget https://gist.githubusercontent.com/rsnk96/87229bd910e01f2ee7c35f96d7cb2f6c/raw/f068812ebd711ed01ebc4c128c8624730ab0dc81/build-zsh.sh -O -)"
```
Comment (rajat2004) [inline 1-BasicSetUp.sh]: Might be better to stick to a release version rather than latest master 
Maybe even the version used by 18.04 (5.4.2)

Comment (rsnk96) [reply 1-BasicSetUp.sh]: Would also have to lock `zimfw` and other `zsh` modules from updating if this is to be implemented, might make things slightly complicated....

**Comment by rsnk96** [comment]
Works on Ubuntu 18.04

**Comment by rsnk96** [comment]
Update instructions for multiple users: probably just need to add
```bash
sudo cp ~/.zshenv /home/$NEW_USER/.zshenv
```
to the code block

**Comment by rajat2004** [comment]
> Works on Ubuntu 18.04

Great, thanks for testing!

> Update instructions for multiple users: probably just need to add
> 
> ```shell
> sudo cp ~/.zshenv /home/$NEW_USER/.zshenv
> ```
> 
> to the code block

What about the `.zimrc`, `.zshrc` etc? Those would also need to be updated right? I haven't installed and checked this myself yet but, the templates are not there in the repo anymore, they're in the script itself, so I think those will also need to be copied from the orignal user's directory itself

File: 1-BasicSetUp.sh (hunk 1 above)
Comment (rajat2004) [reply 1-BasicSetUp.sh]: So specific versions of `zimfw` and `zsh` both?

Comment (rsnk96) [reply 1-BasicSetUp.sh]: Need to lock zimfw updating too

Comment (rsnk96) [reply 1-BasicSetUp.sh]: But we can leave it for now, I think that would be solving a problem which does not yet exist...

Comment (rajat2004) [reply 1-BasicSetUp.sh]: I think we should do it, both `zsh` and `zimfw` versions should be fixed to be ones tested to be working, or at least going by master for both doesn't seem good

File: 1-BasicSetUp.sh [hunk 2]
```diff
@@ -67,15 +67,19 @@ if [[ -d $zsh_folder ]];then
 fi
 
 spatialPrint "Setting up Zsh + Zim now"
-execute sudo apt-get install zsh -y
+# Ubuntu 16.04 has older zsh in repository
+if [[ $(cat /etc/os-release | grep "VERSION_ID" | grep -o -E '[0-9][0-9]' | head -n 1) -le 16 ]]; then
+    zsh_build_script=$(wget https://gist.githubusercontent.com/rsnk96/87229bd910e01f2ee7c35f96d7cb2f6c/raw/f068812ebd711ed01ebc4c128c8624730ab0dc81/build-zsh.sh -O -)
+    # Install specific version of zsh
+    zsh_versioned_script=$(echo $zsh_build_script | sed '/cd zsh/a git checkout zsh-5.8')
```
Comment (rajat2004) [inline 1-BasicSetUp.sh]: Doesn't seem to be a very good way to do it, and doesn't work right now
Should this done only for 16.04 or for 18.04 as well?

File: 1-BasicSetUp.sh (hunk 1 above)
Comment (rajat2004) [reply 1-BasicSetUp.sh]: Right now, `zsh` on source build is the latest release, as for `zimfw`, the script downloads the latest release for that as well, can't change that, other way would be manual install which is complex, and can change easily

**Comment by rajat2004** [comment]
@rsnk96 Ah hell! I force-pushed the branch without fetching your commit for updating the Readme for the multiple users. If possible, could you add back your commit now, sorry for this!

File: 1-BasicSetUp.sh (hunk 1 above)
Comment (rsnk96) [reply 1-BasicSetUp.sh]: Hmmm agreed. Disabling zimfw updates is detailed [here](https://github.com/zimfw/zimfw#settings)

File: 1-BasicSetUp.sh (hunk 2 above)
Comment (rsnk96) [reply 1-BasicSetUp.sh]: Do you think it should be done ofr 18.04 too? I think the apt version of zsh should be stable, so that should be fine...

Comment (rajat2004) [reply 1-BasicSetUp.sh]: Same here, the apt one should be okay for 18.04
The build is working now, so marking this one as resolved

File: 1-BasicSetUp.sh (hunk 1 above)
Comment (rajat2004) [reply 1-BasicSetUp.sh]: Well, yes, we could disable any updates, but maybe that should be upto the user?

**Comment by rajat2004** [comment]
So the main points left to decide on-
1. For source built `zsh`, should we stick to a specific version or the latest tag? (Only on Ubuntu 16.04, on 18.04, `apt` install)
2. Disable `zimfw` from updating or not?

IMO latest tag should be good enough for source built, and disabling `zimfw` from checking updates doesn't seem to be a good thing, the user can decide whether or not to do it, though I don't have a strong preference for any choice

**Comment by rsnk96** [comment]
> IMO latest tag should be good enough for source built, and disabling zimfw from checking updates doesn't seem to be a good thing, the user can decide whether or not to do it, though I don't have a strong preference for any choice

I'm okay with both 👍

**Review by rsnk96 — APPROVED** [review]

**Comment by rajat2004** [comment]
@rsnk96 Great, thanks for the review! Merging

**Pull request merged**